scripts/mdt_build.sh

```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
+
+MDT_DIR=/home/ec2-user/projects/mdt
+MDT_VENV=/home/ec2-user/virtualnev/mdt/bin/activate
+SCRIPT_HOME=/home/ec2-user/scripts
+
+cd $MDT_DIR
+git fetch
+if [[ $(git rev-parse HEAD) == $(git rev-parse @{u}) ]]; then
+    # No new commits
+	echo "`date` - No changes" >> $SCRIPT_HOME/mdt_build.log
+	exit 0
+else
+	echo "`date` - New commit found" >> $SCRIPT_HOME/mdt_build.log
+    # Stop MDT Project
+	$SCRIPT_HOME/mdt_run.sh stop
+
+	# Build project
+	cd $MDT_DIR
+	# Save local changes before update (config, etc.)
+	git stash
+	git pull origin master
+	git stash pop
```

Comment:
"stash pop" will create additional problems, because you'll have 2 sources of truth - local and remote.
just "stash" before "pull" - 👍

Reply:
> "stash pop" will create additional problems, because you'll have 2 sources of truth - local and remote.
> just "stash" before "pull" - 👍

In this case, I want to avoid losing the settings that I did in settings,py file. These settings are specific for the environment where the project is deployed (DB credentials, hostname etc.). Also, I want to avoid hardcoded these settings in the git repository. Thet's why I use stash command to save my settings and avoid conflict when I will pull a new version from git. After pull request,  I use git stash pop to restore my local changes and remove them from the stash. Is it exist a better way to do this?

Reply:
Simplest way to do it right is to add settings file to .gitignore and do not track it in SCM (and maybe even move it to different folder so code can be easily deleted/redeployed)

Here's more inspirations how to separate sensitive settings:
https://code.djangoproject.com/wiki/SplitSettings